### DISCUSSION TITLE
عدم نمایش کارت های کاربر در بانک ملت

### DISCUSSION THREAD

**Discussion by shaahink** [Sina-Soltani/Parbad#219] [Q&A]
<div dir="rtl" markdown="1">
سلام

با معرفی SetMellatMobileNumber در نسخه های اخیر، سعی کردم شماره ی موبایل کاربر را با فرمت خواسته شده در مستندات بانک ملت به درگاه ارسال کنم تا امکان بارگزاری کارت های کاربر و همچنین ذخیره ی کارت کاربر به فرمت خواسته شده را فراهم کنم. اما درگاه نه چک باکس ذخیره و نه کارتهای قبلی را نمایش نمی دهد. آیا راجع به این موضوع اطلاعی دارید؟

نحوه ی استفاده از متد ذکر شده بدین صورت بوده:
<div dir="ltr" markdown="1">
`invoice.SetMellatMobileNumber("989121231111");`
</div>
متشکرم

**Comment by Sina-Soltani** [comment]
درود

این ویژگی مطابق با مستندات خود درگاه ملت پیاده سازی شده
بنابراین بهتره از پشتیبانی درگاه بپرسید مشکل از کجا هست
لطفا در صورت انجام این کار،‌نتیجه رو اینجا بنویسید تا دیگران هم بتونن بخونن

**Reply by shaahink** [reply]
با پشتیبانی تماس گرفتم، گفتند کوچک و بزرگ بودن اسم فیلد رو چک کنید. 
![image](https://user-images.githubusercontent.com/1635787/123106473-8f24bb80-d430-11eb-8cf5-bda4eb41da2a.png)
به نظر کد با مستندات مطابقت نداره. 

https://github.com/Sina-Soltani/Parbad/blob/be4fe84a81722639288d912a596eb899a306da72/src/Parbad/src/Gateway/Mellat/Internal/MellatHelper.cs#L72

آیا تنها روش تست کردن این موضوع تغییر کد و کامپایل مجددش هست؟

**Reply by Sina-Soltani** [reply]
احتمالا همین باید مشکل باشه
هر چند که در مستنداتشون هیچ توضیحی نمیدن که باید این چنین باشه

برای تست بله باید کامپایل بشه

**Reply by Sina-Soltani** [reply]
من یک نسخه بتا درست میکنم که شما میتونید نصب کنید و تست کنید
پس از نصب اگر صحیح بود نسخه اصلی رو منتشر میکنم

بعد از درست شدن نسخه بتا اینجا اطلاع میدم بهتون امروز یا فردا

**Comment by shaahink** [comment]
خیلی عالی
حتما 👍

**Reply by Sina-Soltani** [reply]
لطفا نسخه بتا رو نصب کنید و تست کنید
نتیجه رو اطلاع بدید

`Install-Package Parbad -Version 3.7.1-beta1`

**Reply by shaahink** [reply]
کار کرد، ایول :) 
یک سوالی، به نظرتون میشه تمهیدی تدارک دید که فرمی که به وب سرویس پست میشه رو بشه 
override
کرد؟ 
مثلا اگر بانکی قابلیت جدیدی اضافه کرد، دست توسعه دهنده برای سفارشی کردن فرمی که ارسال میشه باز باشه.

**Reply by Sina-Soltani** [reply]
بله امکانش هست
احتمال داره در نسخه بعدی این قابلیت رو تغییر بدم تا بشه ویرایش کرد
